modules/store/testing/src/testing.ts

```diff
@@ -18,22 +23,140 @@ export interface MockStoreConfig<T> {
   selectors?: MockSelector[];
 }
 
+/**
+ * @description
+ * Creates mock store providers.
+ *
+ * @param config `MockStoreConfig<T>` to provide the values for `INITIAL_STATE` and `MOCK_SELECTORS` tokens.
+ * By default, `initialState` and `selectors` are not defined.
+ * @returns Mock store providers that can be used with both `TestBed.configureTestingModule` and `Injector.create`.
+ *
+ * @usageNotes
+ *
+ * **With `TestBed.configureTestingModule`**
+ *
+ * ```typescript
+ * describe('Books Component', () => {
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     TestBed.configureTestingModule({
+ *       providers: [
+ *         provideMockStore({
+ *           initialState: { books: { entities: [] } },
+ *           selectors: [
+ *             { selector: selectAllBooks, value: ['Book 1', 'Book 2'] },
+ *             { selector: selectVisibleBooks, value: ['Book 1'] },
+ *           ],
+ *         }),
+ *       ],
+ *     });
+ *
+ *     store = TestBed.inject(MockStore);
+ *   });
+ * });
+ * ```
+ *
+ * **With `Injector.create`**
+ *
+ * ```typescript
+ * describe('Counter Component', () => {
+ *   let injector: Injector;
+ *   let store: MockStore;
+ *
+ *   beforeEach(() => {
+ *     injector = Injector.create({
+ *       providers: [
+ *         provideMockStore({ initialState: { counter: 0 } }),
+ *       ],
+ *     });
+ *     store = injector.get(MockStore);
+ *   });
+ * });
+ * ```
+ */
 export function provideMockStore<T = any>(
   config: MockStoreConfig<T> = {}
-): Provider[] {
+): (ValueProvider | ExistingProvider | FactoryProvider)[] {
   setNgrxMockEnvironment(true);
   return [
-    ActionsSubject,
-    MockState,
-    MockStore,
+    {
+      provide: ActionsSubject,
+      useFactory: () => new ActionsSubject(),
```

Comment:
Why not use `useClass` here?

```suggestion
      useClass: ActionsSubject,
```

Reply:
@brandonroberts Because class providers are different for `TestBed.configureTestingModule` and `Injector.create`. `TestBed.configureTestingModule` expects `ClassProvider`. On the other hand, `Injector.create` expects `StaticClassProvider`. However, they both accept `FactoryProvider`.

Reply:
Ok, what about just `{ provide: ActionsSubject, deps: [] }`? I know it's compatible with the Injector, and should be compatible with the TestBed also

Reply:
For some reason the `ActionsSubject` is `undefined` in this case.